src/index.ts

```diff
@@ -11,6 +12,32 @@ export function analytics(config: AnalyticsConfig): AnalyticsEventService {
   return new AnalyticsEventReporter(config);
 }
 
+/**
+ * The exported function of Yext Analytics Events SDK used for Google Tag Manager.
+ * @public
+ */
+export function analyticsGTM(): Promise<string> {
+  const gtmPayload = window['analyticsEventPayload'];
+  let response: Promise<string>;
+  if (gtmPayload) {
+    const config = gtmPayload[0][1] as Record<string, unknown>;
```

Comment:
We are guaranteed that gtmPayload will be length 2. Config will always be inserted, as GTM will not allow for an empty API Key, session tracking, etc. field. For payload, action is a required field so that will not be empty either.